plugins/woocommerce-admin/client/products/components/product-link-field.tsx

```diff
@@ -0,0 +1,93 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Button, Spinner } from '@wordpress/components';
+import { cleanForSlug } from '@wordpress/url';
+import {
+	Product,
+	PRODUCTS_STORE_NAME,
+	WCDataSelector,
+} from '@woocommerce/data';
+import { useFormContext } from '@woocommerce/components';
+import { useSelect } from '@wordpress/data';
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { EditProductLinkModal } from '../shared/edit-product-link-modal';
+import { getProductPermalinkParts } from '../utils/get-product-permalink-parts';
+import { useProductAutoSave } from '../hooks/use-product-auto-save';
+
+export const ProductLinkField = () => {
+	const { errors, touched, values } = useFormContext< Product >();
+	const [ showProductLinkEditModal, setShowProductLinkEditModal ] =
+		useState( false );
+	const hasNameError = Boolean( touched.name ) && Boolean( errors.name );
+	const { isAutoSaving } = useProductAutoSave( [ 'name' ] );
+
+	const { permalinkPrefix, permalinkSuffix, isResolving } = useSelect(
+		( select: WCDataSelector ) => {
+			if ( ! values.id ) {
+				return {
+					permalinkPrefix: null,
+					permalinkSuffix: null,
+				};
+			}
+
+			const { getProduct, isResolving: isProductResolving } =
+				select( PRODUCTS_STORE_NAME );
+			const product = getProduct( values.id );
```

Comment:
Instead of using the `getPermalinkParts` from the data store, I'm manually getting the product because of oddities in the response of `getPermalinkParts`.

I suspect that because [createRegistrySelector](https://github.com/woocommerce/woocommerce/blob/ab0378634a7f01ac99122ea8466eabb51c4c96dd/packages/js/data/src/products/selectors.ts#L125-L156) is used, the response is cached and does not get updated properly when our selector is within a condition.

Reply:
`getPermalinkParts` was just updated to use the products data store (part of an issue I was working on).

https://github.com/woocommerce/woocommerce/pull/36706

Reply:
It feels like we could merge #36706 and then use the `getPermalinkParts` selector? It would be good to reconcile the two PRs, since they are both dealing with the permalink.

Reply:
Left one comment on that PR that could help this work for auto-saved products.